### PR TITLE
Unittests

### PR DESCRIPTION
**Pull request by KEggensperger**
This PR adds two new unittests

1) check whether RNGs are used correctly
2) Evaluate random configurations (using pynisher to terminate long running cfgs)

### CODE REVIEW

File: tests/test_ml/test_init_rng.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import inspect
+import pkgutil
+import os
+import sys
+
+import unittest
+import unittest.mock
+
+import hpolib
+#import hpolib.util.rng_helper
```
Comment (mfeurer) [inline tests/test_ml/test_init_rng.py]: Could you please remove such commented code?

File: tests/test_ml/test_init_rng.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from hpolib.benchmarks.ml.autosklearn_benchmark import AutoSklearnBenchmark, MulticlassClassificationBenchmark
+import numpy as np
+
+
+class TestInitRng(unittest.TestCase):
+
+    @unittest.mock.patch('hpolib.util.rng_helper.get_rng')
+    def test_init_rng(self, rng_mock):
+        rng_mock.side_effect = Exception("Used get_rng")
+        path = os.path.dirname(hpolib.benchmarks.ml.__file__)
```
Comment (mfeurer) [inline tests/test_ml/test_init_rng.py]: Is it intended that this only goes through the `ml` module and not recursively through the whole `hpolib`?

File: tests/test_ml/test_init_rng.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        for _, pkg, _ in pkgutil.iter_modules([path, ]):
+            pkg_name = "hpolib.benchmarks.ml.{:s}".format(pkg)
+            importlib.import_module(pkg_name)
+            mod_name = sys.modules[pkg_name]
+
+            # Find Abstract Benchmark
+            abstract_class = None
+            for name, obj in inspect.getmembers(mod_name,
```
Comment (mfeurer) [inline tests/test_ml/test_init_rng.py]: What happens if there are multiple benchmarks in one files? Maybe have the if statement before as an internal function?

File: tests/test_ml/test_random_config.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                    abstract_class = obj
+                    break
+
+            if abstract_class is not None:
+                for name, obj in inspect.getmembers(mod_name, inspect.isclass):
+                    if issubclass(obj, abstract_class) and \
+                            inspect.isclass(obj) and \
+                            abstract_class in obj.__bases__:
+                        # Make sure to only test correct implementations
+                        print(obj)
```
Comment (mfeurer) [inline tests/test_ml/test_random_config.py:70]: Why is there a print statement?

Comment (KEggensperger) [reply tests/test_ml/test_random_config.py:70]: If this test crashes, there is no information on which module crashes. With a print statement one can see which benchmark crashes.

File: tests/test_ml/test_init_rng.py (hunk 2 above)
Comment (KEggensperger) [reply tests/test_ml/test_init_rng.py]: Yes, synthetic functions are deterministic and should not need a seed

**Review by mfeurer — APPROVED** [review]

**Review by mfeurer — APPROVED** [review]

**Pull request merged**